sql/opt_hints.h

```diff
@@ -516,6 +501,34 @@ class Opt_hints_qb : public Opt_hints
   const Parser::Subquery_hint *subquery_hint= nullptr;
   uint subquery_strategy= SUBS_NOT_TRANSFORMED;
 
+  /**
+    Returns TRUE if the query block has at least one of the hints
+    JOIN_PREFIX(), JOIN_SUFFIX(), JOIN_ORDER()  (but not JOIN_FIXED_ORDER()!)
+  */
+  bool has_join_order_hints() const
+  {
+    return join_order_hints.size() > 0;
+  }
+
+  /*
+    Adds a join order hint to the array if the capacity is not exceeded.
+    Returns FALSE on success,
+            TRUE on failure (capacity exceeded)
+  */
+  bool add_join_order_hint(Parser::Join_order_hint *hint_arg)
+  {
+    if (join_order_hints.size() >= MAX_ALLOWED_JOIN_ORDER_HINTS)
+      return true;
+    join_order_hints.push_back(hint_arg);
+    return false;
+  }
+
+  void print_irregular_hints(THD *thd, String *str) override;
```

Comment:
Does this need to be public?

Reply:
Not really, moved to the private section